test/scripts/parallel-local/main.js

```diff
@@ -0,0 +1,234 @@
+        worker.send({ test, projectName });
+    } else {
+        console.log(`${projectName} found no more tests...`);
+        worker.send('no_more_tests');
+        workersCompleted.add(projectName);
+    }
+};
+
+/**
+ * Logs the exit of worker processes and checks for completion.
+ * @param {number} code - The exit code of the worker process.
+ * @param {string} projectName - The project name.
+ * @returns {void} Does not return a value.
+ */
+const handleWorkerExit = (code, projectName) => {
+    if (code !== 0) {
+        console.error(`Worker ${projectName} exited with code ${code}`);
+    } else {
+        console.log(`Worker ${projectName} completed successfully`);
+    }
+    workersExited.add(projectName);
+    if (workersCompleted.size === numWorkers && workersExited.size === numWorkers) {
+        displayResults();
+    }
+};
+
+/**
+ * Displays the results of all tests by reading specific log files and aggregating results.
+ * It first prints the contents of 'tests.log', calculates totals from 'results.log',
+ * and finally reads 'fails.log' for any failures.
+ * @returns {void} Does not return a value.
+ */
+const displayResults = () => {
+    console.log('\nResults:\n');
+    testSuites.forEach((testSuiteName) => {
+        readLogFiles(testSuiteName, 'tests.log');
+    });
+    console.log('\n');
+
+    let totalPassing = 0;
+    let totalFailing = 0;
+    let totalPending = 0;
+
+    testSuites.forEach((testSuiteName) => {
+        totalPassing += aggregateResults(testSuiteName, 'results.log', 'Passing');
+        totalFailing += aggregateResults(testSuiteName, 'results.log', 'Failing');
+        totalPending += aggregateResults(testSuiteName, 'results.log', 'Pending');
+    });
+
+    // Calculate elapsed time
+    const endTime = new Date();
+    const elapsed = new Date(endTime - startTime);
+    const minutes = elapsed.getUTCMinutes();
+
+    // Display total passing with elapsed time in minutes and seconds
+    console.log('   ', totalPassing, 'Passing', `(${minutes}m)`);
+    console.log('   ', totalFailing, 'Failing');
+    if (totalPending > 0) {
+        console.log('   ', totalPending, 'Pending');
+    }
+
+    console.log('\n');
+
+    testSuites.forEach((testSuiteName) => {
+        readLogFiles(testSuiteName, 'fails.log');
+    });
+
+    cleanupTestDirectories(testSuites);
+};
+
+/**
+ * Reads and logs the contents of a specified log file from the directory of each test suite.
+ * @param {string} testSuiteName - The name of the test suite directory.
+ * @param {string} logFileName - The name of the log file to read.
+ * @returns {void} Does not return a value.
+ */
+const readLogFiles = (testSuiteName, logFileName) => {
+    const testDirectoryPath = path.join('./database/storage/', testSuiteName);
+    try {
+        const files = fs.readdirSync(testDirectoryPath);
+        files.forEach((file) => {
+            if (file === logFileName) {
```

Comment:
Why not directly reading the file using `logFileName` ?

Reply:
Done